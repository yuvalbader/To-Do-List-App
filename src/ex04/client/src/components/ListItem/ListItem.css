.app-container h1,
.pending-tasks {
  font-family: "Russo One", sans-serif;
  font-size: xxx-large;
  color: rgb(248, 203, 70);
  text-shadow: 3px 3px 0 #268d95, -1px -1px 0 #268d95, 1px -1px 0 #268d95,
    -1px 4px 0 #268d95, 1px 1px 0 #268d95;
  text-align: center;
}

.pending-tasks {
  font-size: large;
  margin-left: 19%;
  margin-top: 80px;
  margin-left: 240px;
}

.todo-list-item text {
  font-family: "Russo One", sans-serif;
  font-size: large;
  color: rgb(248, 203, 70);
  text-shadow: 3px 3px 0 #268d95, -1px -1px 0 #268d95, 1px -1px 0 #268d95,
    -1px 4px 0 #268d95, 1px 1px 0 #268d95;
}

.add-new-container,
.todo-list {
  position: relative;
  display: flex;
  flex-direction: row;
  width: 100%;
  max-width: 600px;
  margin: 0 auto;
  border-radius: 2px;
  padding: 1.4rem 2rem 1.6rem;
}

.todo-list {
  display: flex;
  flex-direction: column;
  height: 80vh;
}

.todo-list-item {
  position: relative;
  list-style: none;
  height: 45px;
  line-height: 45px;
  margin-bottom: 8px;
  background-color: rgb(174, 236, 162, 1);
  border-radius: 3px;
  padding: 0 15px;
  cursor: default;
  border-radius: 5px;
  align-items: middle;
  margin-right: 150px;
  margin-top: revert;
}

.delete-task-btn {
  height: fit-content;
  background: rgba(57, 63, 84, 0.407);
  width: auto;
  margin-top: 10px;
}

.todo-list li:hover {
  background: linear-gradient(45deg, white, lightgreen, darkblue);
  background-size: 500% auto;
  animation: gradient 3s linear infinite;
  cursor: pointer;
}

.add-new-container input {
  flex-grow: 1;
  color: #00b09b;
  font-size: 1.8rem;
  line-height: 2.4rem;
  vertical-align: middle;
  border-style: none;
  background: transparent;
  text-align: center;
}

.add-new-container:after {
  content: "";
  position: absolute;
  left: 0px;
  right: 0px;
  bottom: 0px;
  z-index: 999;
  height: 10px;
  border-bottom-left-radius: 2px;
  border-bottom-right-radius: 2px;
  background-position: 0% 0%;
  background: linear-gradient(45deg, darkblue, lightgreen, darkblue);
  background-size: 500% auto;
  animation: gradient 3s linear infinite;
}

@keyframes gradient {
  0% {
    background-position: 0 0;
  }
  100% {
    background-position: 100% 0;
  }
}

.add-new-container button:hover {
  color: darkblue;
  cursor: pointer;
}

.todo-list {
  overflow: auto;
  max-height: 260px;
}

.todo-list-item {
  margin-top: 10;
  margin-bottom: 10;
  align-items: middle;
  margin-right: 150px;
  margin-top: revert;
}
