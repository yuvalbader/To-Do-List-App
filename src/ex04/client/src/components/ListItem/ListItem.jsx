import React from "react";
import { Avatar } from "monday-ui-react-core";
import PropTypes from "prop-types";

import "../../components/ListItem/ListItem.css";

export default function ListItem({
  item,
  handleDoneTask,
  handleDelete,
  handleEdit,
}) {
  return (
    <li className="list-group-item d-flex justify-content-between my-2">
      <Avatar
        size={Avatar.sizes.SMALL}
        src={item.imgUrl}
        type={Avatar.types.IMG}
      ></Avatar>
      <h6
        className={`test   ${
          item.isDone ? "completed-task" : ""
        }`}
      >
        {item.taskContent}
      </h6>
      <div className="todo-icon px-1 ">
        <span
          className={`mx-2 ${item.isDone ? "text-success" : "text-secondary"}`}
          onClick={() => {
            handleDoneTask(item.id);
          }}
        >
          <i
            className={`${
              item.isDone ? "far fa-check-square" : "far fa-square"
            }`}
          />
        </span>
        <span className="mx-2 text-warning" onClick={() => handleEdit()}>
          <i className="fas fa-pen" />
        </span>
        <span
          className="mx-2 text-danger"
          onClick={() => handleDelete(item.id)}
        >
          <i className="fas fa-trash" />
        </span>
      </div>
    </li>
  );
}

ListItem.propTypes = {
  item: PropTypes.object.isRequired,
  handleDoneTask: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired,
  handleEdit: PropTypes.func.isRequired,
};
