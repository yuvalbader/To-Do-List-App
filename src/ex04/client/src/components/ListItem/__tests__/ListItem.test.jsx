import renderer from "react-test-renderer";
import ListItem from "../ListItem";
import { store } from "../../../Redux/store";

const task = {
  id: "1",
  taskContent: "catch picatch",
  isDone: false,
  imgUrl:
    "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png",
};

test("Render correctly listItem of this task", () => {
  const tree = renderer.create(<ListItem store={store} item={task} />).toJSON();
  expect(tree).toMatchSnapshot();
});
