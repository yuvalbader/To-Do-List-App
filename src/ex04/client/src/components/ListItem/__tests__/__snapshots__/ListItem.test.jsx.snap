// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly listItem of this task 1`] = `
<div>
  <li
    className="list-group-item d-flex justify-content-between my-2"
    data-testid="item-1"
  >
    <div
      className="monday-style-avatar monday-style-avatar--small"
      style={Object {}}
    >
      <div
        aria-hidden={false}
        className="monday-style-avatar_circle monday-style-avatar_circle--img"
        style={Object {}}
        tabIndex={0}
      >
        <img
          className="monday-style-avatar-content_img monday-style-avatar-content_img--small"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png"
        />
      </div>
    </div>
    <h6
      className="test   "
    >
      catch picatch
    </h6>
    <div
      className="todo-icon px-1 "
    >
      <span
        className="mx-2 text-secondary"
        onClick={[Function]}
      >
        <i
          className="far fa-square"
        />
      </span>
      <span
        className="mx-2 text-warning"
      >
        <i
          className="fas fa-pen"
        />
      </span>
      <span
        className="mx-2 text-danger"
        onClick={[Function]}
      >
        <i
          className="fas fa-trash"
        />
      </span>
    </div>
  </li>
</div>
`;
